From 3b7fd5d94b3e13d3ce22b792838609ed91dba8c5 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Thu, 10 May 2018 14:21:27 +0800
Subject: [PATCH 3766/5242] MLK-18265-4 ARM64: dts: freescale: imx8qm: correct
 jpeg power domain

commit  de77e0fa48b236a1862da3e8195d7def6ff87a55 from
https://source.codeaurora.org/external/imx/linux-imx.git

Correct JPEG power domain tree to make sure kernel can
boot up successfully with XRDC enforcement enabled.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Tested-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi      |   24 ++++++++++++++++----
 1 file changed, 20 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index b5a1882..5647a48 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -998,16 +998,32 @@
 				power-domains =<&pd_isi_ch0>;
 			};
 
-			pd_jpgdec: PD_IMAGING_JPEG_DEC {
-				reg = <SC_R_MJPEG_DEC_S0>;
+			pd_jpeg_dec_mp: PD_JPEG_DEC_MP {
+				reg = <SC_R_MJPEG_DEC_MP>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_isi_ch0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_jpgdec: PD_IMAGING_JPEG_DEC {
+					reg = <SC_R_MJPEG_DEC_S0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_jpeg_dec_mp>;
+				};
 			};
 
-			pd_jpgenc: PD_IMAGING_JPEG_ENC {
-				reg = <SC_R_MJPEG_ENC_S0>;
+			pd_jpeg_enc_mp: PD_JPEG_ENC_MP {
+				reg = <SC_R_MJPEG_ENC_MP>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_isi_ch0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_jpgenc: PD_IMAGING_JPEG_ENC {
+					reg = <SC_R_MJPEG_ENC_S0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_jpeg_enc_mp>;
+				};
 			};
 		};
 
-- 
1.7.9.5

